### PR TITLE
Ruler Experimental API: Fix rule group validation

### LINKED ISSUE #2942: Ruler API should not accept invalid rules files
The Ruler API that allows users to write rules files to Cortex should return an error to users who try to upload invalid rules files, for example a rule group with no name.

cc @gotjosh

### PR DESCRIPTION
**Pull request by gotjosh**
**What this PR does**:

Fixed a few cases where the API would allow us to submit a bad rule group payload which would end up overwriting existing ones.

In our case, this is pretty egregious given we operate against individual rule groups you could end up in a situation where you have overwritten a working rule group by mistake. 

**Which issue(s) this PR fixes**:
Fixes #2942 

**Checklist**
- [x] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/ruler/api.go [hunk 1]
```diff
@@ -443,10 +444,13 @@ func (r *Ruler) CreateRuleGroup(w http.ResponseWriter, req *http.Request) {
...
 	errs := r.manager.ValidateRuleGroup(rg)
 	if len(errs) > 0 {
+		e := []string{}
 		for _, err := range errs {
 			level.Error(logger).Log("msg", "unable to validate rule group payload", "err", err.Error())
+			e = append(e, err.Error())
 		}
-		http.Error(w, errs[0].Error(), http.StatusBadRequest)
+
+		http.Error(w, strings.Join(e, ","), http.StatusBadRequest)
```
Comment (jtlisi) [inline pkg/ruler/api.go:453]:
```suggestion
		http.Error(w, strings.Join(e, ", "), http.StatusBadRequest)
```

May be worth adding a space so the errors aren't one continues string

**Review by jtlisi — APPROVED** [review]
LGTM modulo linter

File: pkg/ruler/api.go (hunk 1 above)
Comment (gotjosh) [reply pkg/ruler/api.go]: good shout.

**Review by pracucci — APPROVED** [review]
LGTM, thanks!

**Pull request merged**